jerry-core/ecma/operations/ecma-regexp-object.cpp

```diff
@@ -1261,20 +1261,21 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
 
   MEM_DEFINE_LOCAL_ARRAY (input_buffer_p, input_string_size, lit_utf8_byte_t);
 
+  re_matcher_ctx_t re_ctx;
   ssize_t sz = ecma_string_to_utf8_string (input_string_p, input_buffer_p, (ssize_t) input_string_size);
   JERRY_ASSERT (sz >= 0);
 
   lit_utf8_byte_t *input_curr_p = input_buffer_p;
+  re_ctx.input_start_p = input_buffer_p;
```

Comment:
if I'd like to be picky -- which I'm obviously not :) -- then I'd suggest moving this in the else branch of `if (!input_string_size)`.